modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentFolderResourceImpl.java

```diff
@@ -402,6 +447,44 @@ private Page<DocumentFolder> _getDocumentFoldersPage(
 					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
 	}
 
+	private SPIRatingResource<Rating> _getSPIRatingResource() {
+		return new SPIRatingResource<>(
+			DLFolder.class.getName(), _ratingsEntryLocalService,
+			ratingsEntry -> {
+				Folder folder = _dlAppService.getFolder(
+					ratingsEntry.getClassPK());
+
+				return RatingUtil.toRating(
+					HashMapBuilder.put(
+						"create",
+						addAction(
+							ActionKeys.VIEW, folder.getPrimaryKey(),
+							"postDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).put(
+						"delete",
+						addAction(
+							ActionKeys.VIEW, folder.getPrimaryKey(),
+							"deleteDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).put(
+						"get",
+						addAction(
+							ActionKeys.VIEW, folder.getPrimaryKey(),
+							"getDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).put(
+						"replace",
+						addAction(
+							ActionKeys.VIEW, folder.getPrimaryKey(),
+							"putDocumentFolderMyRating", folder.getUserId(),
+							DLFolder.class.getName(), folder.getGroupId())
+					).build(),
+					_portal, ratingsEntry, _userLocalService);
```

Comment:
Would it be useful to create a private method like this?

```java
	private Map<String, String> _addViewAction(Folder folder, String methodName) {
		return addAction(
			ActionKeys.VIEW, folder.getPrimaryKey(),
			methodName, folder.getUserId(),
			DLFolder.class.getName(), folder.getGroupId());
	}
```

Reply:
I'm following existing patterns, so I'm not sure about this; If you want to give it a try, we can send it in a different PR so we don't delay this.